homeassistant/components/tolosauna/climate.py

```diff
@@ -0,0 +1,225 @@
+) -> None:
+    """Set up climate controls for TOLO Sauna."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]["coordinator"]
+    await coordinator.async_config_entry_first_refresh()
+    async_add_entities([SaunaClimate(coordinator, entry)])
+
+
+class SaunaClimate(ToloSaunaCoordinatorEntity, ClimateEntity):
+    """Sauna climate control."""
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique ID."""
+        return f"{self._config_entry.entry_id}_climate"
+
+    @property
+    def name(self) -> str:
+        """Return name."""
+        return "Sauna Climate"
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return temperature uni."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature_step(self) -> int:
+        """Return target temperature step."""
+        return 1
+
+    @property
+    def precision(self) -> float:
+        """Return precision."""
+        return PRECISION_WHOLE
+
+    @property
+    def current_temperature(self) -> int:
+        """Return current temperature."""
+        return self.status.current_temperature
+
+    @property
+    def current_humidity(self) -> int:
+        """Return current humidity."""
+        return self.status.current_humidity
+
+    @property
+    def target_temperature(self) -> int:
+        """Return target temperature."""
+        return self.settings.target_temperature
+
+    @property
+    def target_humidity(self) -> int:
+        """Return target humidity."""
+        return self.settings.target_humidity
+
+    @property
+    def max_temp(self) -> int:
+        """Return maximum supported temperature."""
+        return DEFAULT_MAX_TEMP
+
+    @property
+    def min_temp(self) -> int:
+        """Return minimum supported temperature."""
+        return DEFAULT_MIN_TEMP
+
+    @property
+    def max_humidity(self) -> int:
+        """Return maximum supported humidity."""
+        return DEFAULT_MAX_HUMIDITY
+
+    @property
+    def min_humidity(self) -> int:
+        """Return minimum supported humidity."""
+        return DEFAULT_MIN_HUMIDITY
+
+    @property
+    def hvac_mode(self) -> str:
+        """Get current HVAC mode."""
+        if self.status.power_on:
+            return HVAC_MODE_HEAT
+        else:
+            return HVAC_MODE_OFF
+
+    @property
+    def hvac_action(self) -> Optional[str]:
+        """Execute HVAC action."""
+        if self.status.calefaction == Calefaction.HEAT:
+            return CURRENT_HVAC_HEAT
+        elif self.status.calefaction == Calefaction.KEEP:
+            return CURRENT_HVAC_IDLE
+        elif self.status.calefaction == Calefaction.INACTIVE:
+            if self.status.fan_on:
+                return CURRENT_HVAC_DRY
+            else:
+                return CURRENT_HVAC_OFF
+        return None
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """Return supported HVAC modes."""
+        return [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_DRY]
```

Comment:
Use attributes instead of properties for constants to make the code more readable.
Docs: https://developers.home-assistant.io/docs/core/entity/#entity-class-or-instance-attributes
Climate:
https://github.com/home-assistant/core/blob/4e1e7a4a718396ccbdaa37b17c6f6410d12528c8/homeassistant/components/climate/__init__.py#L182-L205

Reply:
Done, thx for the hint!